runtime/src/bucket_map_holder.rs

```diff
@@ -9,13 +9,37 @@ use std::time::Duration;
 // will eventually hold the bucket map
 #[derive(Debug, Default)]
 pub struct BucketMapHolder<T: IsCached> {
-    pub stats: BucketMapHolderStats,
+    pub stats: Arc<BucketMapHolderStats>,
+    pub buckets: Vec<Arc<CachedBucket<T>>>,
+}
+
+#[derive(Debug, Default)]
+pub struct CachedBucket<T: IsCached> {
+    // soon, this will hold the hashmap itself: pub map: RwLock<HashMap<Pubkey, AccountMapEntry<T>>>,
+    bin: usize,
+    stats: Arc<BucketMapHolderStats>,
     _phantom: std::marker::PhantomData<T>,
 }
 
+impl<T: IsCached> CachedBucket<T> {
+    pub fn new(stats: Arc<BucketMapHolderStats>, bin: usize) -> Self {
+        Self {
+            bin,
+            stats,
+            ..Self::default()
+        }
+    }
+}
+
 impl<T: IsCached> BucketMapHolder<T> {
-    pub fn new() -> Self {
-        Self::default()
+    pub fn new() -> Arc<Self> {
+        let bins = 0;
+        let stats = Arc::new(BucketMapHolderStats::default());
+        let buckets = (0..bins)
+            .into_iter()
+            .map(|bin| Arc::new(CachedBucket::new(stats.clone(), bin)))
+            .collect();
+        Arc::new(Self { stats, buckets })
     }
```

Comment:
As a user, I would be surprised if I called `new()` on a type `T` and it didn't return `T`, but instead `Arc<T>`. Does the Arc need to be here?

Reply:
Yes, I can clean that up. I changed direction and that was an artifact left over.